### PR TITLE
sync sidecar indexPattern from main config

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

This PR allows user to configure es index pattern  added in common config and is synced to airflow deployments and curator config without any additional effort

## Related Issues

https://github.com/astronomer/issues/issues/5594

## Related PRs

https://github.com/astronomer/houston-api/pull/1521

## Testing

Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging

Do not merge this PR until it lists which release branches this PR should be merged / cherry-picked into.

### CODE REVIEW

**Comment by pgvishnuram** [comment]
@rishkarajgi and @danielhoherd

File: charts/elasticsearch/templates/_helpers.tpl [hunk 1]
```diff
@@ -153,3 +153,12 @@ imagePullSecrets:
 {{ . }},
 {{- end -}}
 {{- end -}}
+
+
+{{- define "curator.indexPattern" -}}
+{{ if and .Values.global.loggingSidecar.enabled  .Values.global.loggingSidecar.indexPattern }}
+{{- .Values.global.loggingSidecar.indexPattern | squote }}
+{{ else }}
+{{- .Values.curator.age.timestring | squote}}
```
Comment (danielhoherd) [inline charts/elasticsearch/templates/_helpers.tpl:162]: Why do we need "squote" here? We'll definitely need to make test cases with various quoting possibilities. https://helm.sh/docs/chart_template_guide/function_list/#printf may be needed.

**Review by danielhoherd — COMMENTED** [review]
Looks good. I think that having tests will make it all more clear.

File: charts/elasticsearch/templates/curator/es-curator-configmap.yaml [hunk 2]
```diff
@@ -34,7 +34,7 @@ data:
         - filtertype: age
           source: name
           direction: older
-          timestring: '{{ .Values.curator.age.timestring }}'
+          timestring: {{ template "curator.indexPattern" . }}
```
Comment (pgvishnuram) [inline charts/elasticsearch/templates/curator/es-curator-configmap.yaml:37]: @danielhoherd it was with single quote thats why i added the same in the helper templates

**Review by rishkarajgi — APPROVED** [review]
lgtm, please address @danielhoherd's comments

File: tests/chart_tests/test_houston_configmap.py [hunk 3]
```diff
@@ -232,6 +232,42 @@ def test_houston_configmap_with_loggingsidecar_enabled_with_overrides():
...
+    doc = docs[0]
+    prod_yaml = yaml.safe_load(doc["data"]["production.yaml"])
+    log_cmd = 'log_cmd = " 1> >( tee -a /var/log/sidecar-log-consumer/out.log ) 2> >( tee -a /var/log/sidecar-log-consumer/err.log >&2 ) ; "'
+    assert (
+        log_cmd in prod_yaml["deployments"]["helm"]["airflow"]["airflowLocalSettings"]
+    )
+    assert prod_yaml["deployments"]["loggingSidecar"] == {
+        "enabled": True,
+        "name": sidecar_container_name,
+        "image": "quay.io/astronomer/ap-vector:0.22.3",
```
Comment (danielhoherd) [inline tests/chart_tests/test_houston_configmap.py:265]: For test cases we should always set the test to something that is obviously not real data unless parsing is necessary. This makes debugging errors easier.

```suggestion
    image_name = "example.com/some-repo/test-image-name:test-tag-foo"
    indexPattern = "%Y.%m"
    docs = render_chart(
        values={
            "global": {
                "loggingSidecar": {
                    "enabled": True,
                    "name": sidecar_container_name,
                    "image": image_name,
                    "indexPattern": indexPattern,
                }
            }
        },
        show_only=["charts/astronomer/templates/houston/houston-configmap.yaml"],
    )

    common_test_cases(docs)
    doc = docs[0]
    prod_yaml = yaml.safe_load(doc["data"]["production.yaml"])
    log_cmd = 'log_cmd = " 1> >( tee -a /var/log/sidecar-log-consumer/out.log ) 2> >( tee -a /var/log/sidecar-log-consumer/err.log >&2 ) ; "'
    assert (
        log_cmd in prod_yaml["deployments"]["helm"]["airflow"]["airflowLocalSettings"]
    )
    assert prod_yaml["deployments"]["loggingSidecar"] == {
        "enabled": True,
        "name": sidecar_container_name,
        "image": image_name,
```

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Looks mostly good. I left an optional suggestion to improve a test case. I am also requesting that we parametrize the indexPattern test to make sure `squote` is compatible with index patters that have single quotes. This is highly unlikely to be a real world configuration, but we need to make sure we are properly quoting our data and not making it possible to produce invalid yaml.

File: charts/elasticsearch/templates/curator/es-curator-configmap.yaml (hunk 2 above)
Comment (pgvishnuram) [reply charts/elasticsearch/templates/curator/es-curator-configmap.yaml:37]: pytest doen;t seems to like % in a string that gets interpolated

File: tests/chart_tests/test_houston_configmap.py [hunk 4]
```diff
@@ -232,6 +232,42 @@ def test_houston_configmap_with_loggingsidecar_enabled_with_overrides():
     assert "vector" in prod_yaml["deployments"]["loggingSidecar"]["image"]
 
 
+def test_houston_configmap_with_loggingsidecar_enabled_with_indexPattern():
+    """Validate the houston configmap and its embedded data with
+    loggingSidecar."""
+    sidecar_container_name = "sidecar-log-test"
+    image_name = "example.com/some-repo/test-image-name:test-tag-foo"
+    indexPattern = "%Y.%m"
```
Comment (danielhoherd) [inline tests/chart_tests/test_houston_configmap.py:240]: This parametrize block works and shows that the single-quote val breaks the helm mechanic.
```suggestion
@pytest.mark.parametrize(
    "indexPattern",
    [
        "%Y.%m",
        "%Y.%m.%d",
        "a-single-quote-'-in-the-indexPattern",
        'a-double-quote-"-in-the-indexPattern',
    ],
)
def test_houston_configmap_with_loggingsidecar_enabled_with_indexPattern():
    """Validate the houston configmap and its embedded data with
    loggingSidecar."""
    sidecar_container_name = "sidecar-log-test"
    image_name = "example.com/some-repo/test-image-name:test-tag-foo"
```

Comment (danielhoherd) [reply tests/chart_tests/test_houston_configmap.py:240]: Another way we could work around this is to add a comment that single-quotes are not allowed. That would be a really easy non-code way to handle this potential problem.

Comment (danielhoherd) [reply tests/chart_tests/test_houston_configmap.py:240]: I tried to find a way to handle both single and double quotes and I don't see an easy way to do it without making really ugly code. Helm printf does not appear to have any mechanism to produce the right string. I think we can skip merging the parametrize change and just put a comment in the values.yaml that quote characters are not supported in the indexPattern.

**Comment by pgvishnuram** [comment]
it will require a houston update to pass the indexPattern config

**Pull request merged**